src/rust-crypto/libolm_migration.ts

```diff
@@ -158,7 +160,17 @@ async function migrateBaseData(
     // If we have a backup recovery key, we need to try to figure out which backup version it is for.
     // All we can really do is ask the server for the most recent version.
     if (recoveryKey) {
-        const backupInfo = await requestKeyBackupVersion(http);
+        let backupCallDone = false;
+        let backupInfo: KeyBackupInfo | null = null;
+        while (!backupCallDone) {
```

Comment:
What happens if we are offline or something like that?

Reply:
I.e. should we give up eventually?

Reply:
As discussed with @richvdh, we might as well [block the application until it works](https://github.com/matrix-org/matrix-js-sdk/pull/4090#discussion_r1517805018).
If we want to give up we would need to catch such errors in `LifeCycle#handleLoadSessionFailure` and is probably part of a bigger task for proper support of offline mode on web

Reply:
Not planned for now, needs design and some refactoring